test/integration/plugins/solidfire/TestManagedSystemVMs.py

```diff
@@ -0,0 +1,585 @@
+        virtual_machine.delete(self.apiClient, True)
+
+        virtual_router = list_routers(self.apiClient, listall=True, state="Running")[0]
+
+        system_vms.append(virtual_router)
+
+        self._check_system_vms(system_vms, primary_storage_1.id)
+
+        primary_storage[TestData.name] = TestData.get_name_for_solidfire_storage()
+
+        primary_storage_2 = StoragePool.create(
+            self.apiClient,
+            primary_storage
+        )
+
+        StoragePool.enableMaintenance(self.apiClient, primary_storage_1.id)
+
+        self._wait_for_storage_cleanup_thread(system_vms)
+
+        sf_util.purge_solidfire_volumes(self.sf_client)
+
+        system_vms = self._wait_for_and_get_running_system_vms(2)
+
+        virtual_router = list_routers(self.apiClient, listall=True, state="Running")[0]
+
+        system_vms.append(virtual_router)
+
+        self._check_system_vms(system_vms, primary_storage_2.id)
+
+        StoragePool.cancelMaintenance(self.apiClient, primary_storage_1.id)
+
+        primary_storage_1.delete(self.apiClient)
+
+        self._disable_zone_and_delete_system_vms(virtual_router)
+
+        self._wait_for_storage_cleanup_thread(system_vms)
+
+        sf_util.purge_solidfire_volumes(self.sf_client)
+
+        primary_storage_2.delete(self.apiClient)
+
+        self._verify_no_active_solidfire_volumes()
+
+        self._prepare_to_stop_using_managed_storage_for_system_vms()
+
+        self.zone.update(self.apiClient, id=self.zone.id, allocationstate=enabled)
+
+        self._wait_for_and_get_running_system_vms(2)
+
+    @attr(hypervisor='XenServer')
+    def test_02_failure_to_create_service_offering_with_customized_iops(self):
+        try:
+            ServiceOffering.create(
+                self.apiClient,
+                self.testdata[TestData.systemOfferingFailure]
+            )
+
+            self.assert_(True, "The service offering was created, but should not have been.")
+        except:
+            pass
+
+    def _prepare_to_use_managed_storage_for_system_vms(self):
+        self._update_system_vm_unique_name(TestManagedSystemVMs._secondary_storage_unique_name, TestManagedSystemVMs._secondary_storage_temp_unique_name)
+        self._update_system_vm_unique_name(TestManagedSystemVMs._console_proxy_unique_name, TestManagedSystemVMs._console_proxy_temp_unique_name)
+        self._update_system_vm_unique_name(TestManagedSystemVMs._virtual_router_unique_name, TestManagedSystemVMs._virtual_router_temp_unique_name)
+
+        self._update_system_vm_unique_name_based_on_uuid(self.secondary_storage_offering.id, TestManagedSystemVMs._secondary_storage_unique_name)
+        self._update_system_vm_unique_name_based_on_uuid(self.console_proxy_offering.id, TestManagedSystemVMs._console_proxy_unique_name)
+        self._update_system_vm_unique_name_based_on_uuid(self.virtual_router_offering.id, TestManagedSystemVMs._virtual_router_unique_name)
```

Comment:
Should we assert after each of these `self._update_system_vm`* calls to verify that they completed successfully?